realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1657,14 +1657,24 @@ public static void migrateRealm(RealmConfiguration configuration, @Nullable Real
     }
 
     /**
-     * Deletes the Realm file specified by the given {@link RealmConfiguration} from the filesystem.
+     * Deletes the Realm file along with the related temporary files specified by the given {@link RealmConfiguration}
+     * from the filesystem. Temporary file with ".lock" extension won't be deleted.
```

Comment:
is Core responsible for deleting the lock file? and when?

Reply:
no, lock file should never be deleted. There is no reliable way to delete it for multi processes case.